plugin/src/test/java/org/jenkinsci/plugins/workflow/cps/CpsBodyExecutionTest.java

```diff
@@ -273,4 +277,35 @@ public static class Execution extends AbstractStepExecutionImpl {
         });
     }
 
+    @Test public void failureWithNoContextVars() throws Throwable {
+        rr.then(r -> {
+            WorkflowJob p = r.createProject(WorkflowJob.class, "p");
+            p.setDefinition(new CpsFlowDefinition("noOpBlock { throw new Exception('oops') }", true));
```

Comment:
I think the Pipeline @gbhat618 was using just had an outermost `parallel` step, and that was enough to cause issues, since it doesn't insert any context variables. `stage` is actually fine because it uses `EnvironmentExpander`. I think various steps would have similar problems to `parallel`, such as `timeout`, but for test stability it seemed clearest to create a synthetic step in case the real steps add context variables in the future.